app/uk/gov/hmrc/incorporatedentityidentification/services/ValidateIncorporatedEntityDetailsService.scala

```diff
@@ -26,14 +26,13 @@ import scala.concurrent.{ExecutionContext, Future}
 @Singleton
 class ValidateIncorporatedEntityDetailsService @Inject()(getCtReferenceConnector: GetCtReferenceConnector)(implicit ec: ExecutionContext) {
 
-  def validateDetails(companyNumber: String, ctutr: String)(implicit hc: HeaderCarrier): Future[IncorporatedEntityDetailsValidationResult] = {
+  def validateDetails(companyNumber: String, optCtUtr: Option[String])(implicit hc: HeaderCarrier): Future[IncorporatedEntityDetailsValidationResult] = {
     getCtReferenceConnector.getCtReference(companyNumber).map {
-      case Some(retrievedCtutr) if retrievedCtutr == ctutr =>
-        DetailsMatched
-      case Some(_) =>
-        DetailsMismatched
-      case None =>
-        DetailsNotFound
+      case Some(retrievedCtUtr) => optCtUtr match {
+        case Some(ctUtr) => if(retrievedCtUtr == ctUtr) DetailsMatched else DetailsMismatched
```

Comment:
I am always against nested if. ..... if you want we can use what scala provides already with the case statement to check if retrievedCtUtr  == ctUtr ... otherwise not problem for me.
<img width="1278" alt="Screenshot 2022-03-15 at 10 50 11" src="https://user-images.githubusercontent.com/4958044/158362017-b6d3f5bd-554b-4e39-a3ca-b3082793150f.png">

Reply:
I can see I'm dealing with a Scala purist. I've updated the logic as requested.